cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/InvalidRepositoryConfigException.java

```diff
@@ -37,6 +37,11 @@ public InvalidRepositoryConfigException(String message) {
     this.failures = Collections.unmodifiableList(new ArrayList<>());
   }
 
+  public InvalidRepositoryConfigException(String message, Exception cause) {
```

Comment:
All lgtm. Just a nit, wondering if we should rename this Exception to something like `RepositoryConfigValidationException`, which both takes care of the format errors for static fields and runtime validation errors. Cause even valid config can throw unexpected exceptions like network timeout issue? The name `InvalidRepositoryConfig...` could be misleading somehow.


Also before merge, pls squash commits, thanks!

Reply:
Renamed the exception. I'm waiting for an approval from @albertshau or @chtyim before squashing commits just in case I need to revert.